src/io/mod.rs

```diff
@@ -7,3 +10,18 @@ pub use unix::IoDriver;
 
 #[cfg(not(target_family = "unix"))]
 compile_error!("Specified target platform is not supported (only unix family supported)");
+
+pub(crate) enum WritableData {
+    Single(Bytes),
+    Double(Bytes, Bytes),
+}
+
+pub(crate) trait WritableDataCreator<R>:
+    FnOnce(u64) -> Result<(WritableData, R)> + Send + 'static
```

Comment:
I don't like the approach with delegate. It is better to declare the function within this trait and call it